tests/common/lease_test.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2022 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"go.etcd.io/etcd/tests/v3/framework/config"
+	"go.etcd.io/etcd/tests/v3/framework/testutils"
+)
+
+func TestLeaseGrantTimeToLive(t *testing.T) {
+	testRunner.BeforeTest(t)
+
+	tcs := []struct {
+		name   string
+		config config.ClusterConfig
+	}{
+		{
+			name:   "NoTLS",
+			config: config.ClusterConfig{ClusterSize: 1},
+		},
+		{
+			name:   "PeerTLS",
+			config: config.ClusterConfig{ClusterSize: 3, PeerTLS: config.ManualTLS},
+		},
+		{
+			name:   "PeerAutoTLS",
+			config: config.ClusterConfig{ClusterSize: 3, PeerTLS: config.AutoTLS},
+		},
+		{
+			name:   "ClientTLS",
+			config: config.ClusterConfig{ClusterSize: 1, ClientTLS: config.ManualTLS},
+		},
+		{
+			name:   "ClientAutoTLS",
+			config: config.ClusterConfig{ClusterSize: 1, ClientTLS: config.AutoTLS},
+		},
+	}
+	for _, tc := range tcs {
+		t.Run(tc.name, func(t *testing.T) {
+			clus := testRunner.NewCluster(t, tc.config)
+			defer clus.Close()
+			cc := clus.Client()
+
+			testutils.ExecuteWithTimeout(t, 10*time.Second, func() {
+				leaseResp, err := cc.Grant(10)
+				require.NoError(t, err)
+
+				ttlResp, err := cc.TimeToLive(leaseResp.ID, config.LeaseOption{})
+				require.NoError(t, err)
```

Comment:
I don't know testify enough to say which one is better, however in most of the code we use `assert` package. I think we should keep it consistent, however I'm interested if there is any advantage to use `require` instead.

Reply:
`require` will end the test if an assertion isn't met, but `assert` will continue the test and stack the failure reasons. When something is a precondition it's much easier to understand why something failed when using `require` (and avoids potential panics when dealing with pointers)

Reply:
Interesting, I assumed that this is how `assert` behaved :P (naming is hard). Might be worth considering to switch. 

Would it be ok for you to use `assert` for now to keep it consistent with rest of the code?
I think it would make sense to create a issue with proposal to switch to `require` and migrate whole repo at once.

Reply:
@serathius In this case i'd need to switch to using `t.Fatalf` and handwritten comparisons instead - otherwise tests will panic if anything ever fails, but can do that if desirable.

Reply:
Resolved everything else, leaving this one open for now because the re-write of all the "i need the test to stop to be safe" assertions will take a while.

Reply:
Will do 👍  (fwiw i'll probably have migrated many of the existing e2es by next week at the pace I'm going, so might end up doing a lot of migration at the same time)

Reply:
Great to hear this. I think it would also be great if you looked into https://github.com/etcd-io/etcd/issues/13167. We should not only do the migration, but also monitor the improvement over time.